src/lib/tabs/tab-nav-bar/tab-nav-bar.ts

```diff
@@ -37,10 +41,14 @@ export class MdTabNavBar implements AfterContentInit, OnDestroy {
   /** Combines listeners that will re-align the ink bar whenever they're invoked. */
   private _realignInkBar: Subscription = null;
 
+  /** Subject that emits when the component has been destroyed. */
+  private _onDestroy = new Subject<void>();
```

Comment:
Instead of having an internal `Subject`, why not save the subscription and unsubscribe on destroy instead? On a related note, I wonder whether the `_tabLinks` wouldn't get completed automatically on destroy.